integration/src/main/java/ai/djl/integration/tests/pytorch/MkldnnTest.java

```diff
@@ -11,19 +11,24 @@
  * and limitations under the License.
  */
 
-package integration;
+package ai.djl.integration.tests.pytorch;
 
+import ai.djl.integration.util.TestUtils;
 import ai.djl.ndarray.NDArray;
 import ai.djl.ndarray.NDManager;
 import ai.djl.ndarray.types.Shape;
 import java.util.Arrays;
+import org.testng.SkipException;
 import org.testng.annotations.Test;
 
-/** The file is for testing PyTorch MKLDNN functionalities. */
+/** The package is specific for PyTorch integration test. */
 public class MkldnnTest {
 
     @Test
-    public void testMKLDNNForPyTorch() {
+    public void testMKLDNN() {
```

Comment:
Is there a reason you want to move this? I think it makes more sense to leave the integration module as focusing on the engine requirements of the DJL spec itself while tests that focus on engine specific things can be in their separate folders

Reply:
we found PR #212 ran into a situation where it uses old JNI to run the test and obviously failed. By moving this to integration test, we will rebuild the JNI first then run it on `Continuous build` task

Reply:
Is there a way we can check for stale JNI from gradle? Then, we can choose whether to download or compile JNI dynamically during the gradle build

Reply:
There are a few options.
1. check files change in those C++ code and java "header"
2. just recompile the jni one way or the other

Reply:
In addition, one of the advantage of putting pytorch specific tests in integrations is we can just run integration test for verification when we do PyTorch version upgrade

Reply:
Discussed with Frank offline, the windows JNI build is pretty slow. We want to keep the test fast.

Reply:
I am thinking something closer to 1. This isn't the first time the JNI caching has caused some weird problems.

Here is what I am thinking. Can we move the c tests to an isolated folder like `/pytorch/pytorch-native/src/main/jni` so we can hash the folder contents with something like `tar cf - . | gsha1sum`. Then, we upload the hash when we built the jni and check during download to know whether we use the hashed version or should recompile.

Reply:
I would say no to abandon Windows JNI build. Historically there are cases JNI passed in Mac and failed on Windows. It took me 4 days to fix them on Windows

Reply:
For #212, we can merge it without passing the tests to unblock the PR